### PR TITLE
Enable support for custom Form Builder Styling with the new Neos UI

### PR DESCRIPTION
**Pull request by bwaidelich**
This adjusts the content element wrapping of all node based forms
and form elements so they have a custom attribute `data-_neos-form-builder-type`
with a value of the corresponding node type.
In addition empty content collections get a `data-_neos-form-builder-empty-collection`
attribute.

This allows for custom styling of the builder in the ReactJS based
Neos backend as described in the README.

Resolves: #24

### CODE REVIEW

**Comment by bwaidelich** [comment]
*Note:* For this to work as expected, neos/neos-ui#1648 has to be merged

**Comment by Sebobo** [comment]
And https://github.com/neos/neos-development-collection/pull/1919 needs to be applied.

**Comment by Sebobo** [comment]
Rendering also has some other issues when using the fusion forms preset.

File: Resources/Private/Styles/scss/Includes/_Element.scss [hunk 1]
```diff
@@ -1,18 +1,16 @@
-#{select("Neos.Form.Builder:ElementCollection")} {
-  margin-top: 30px;
+[data-_neos-form-builder-type="Neos.Form.Builder:ElementCollection"] {
+    margin-top: 30px;
+    & > [data-_neos-form-builder-type] {
+        position: relative;
```
Comment (Sebobo) [inline Resources/Private/Styles/scss/Includes/_Element.scss]: The default form template and the fusion template have a wrapping `div.clearfix` which is between the `ElementCollection` and the element with the attribute. This causes this selector to not apply.

**Comment by bwaidelich** [comment]
@Sebobo Thanks again for your review. There are multiple issues:

> There seems to be a problem with sections, they render the element wrapping of elements twice.

This is indeed a bug, but most probably not related to this change. I created a separate ticket: #27
Unfortunately it wasn't as easy to solve as I hoped for..

> The default form template and the fusion template have a wrapping div.clearfix which is
> between the ElementCollection and the element with the attribute

I could not reproduce this with the default preset (= Fluid rendering) but it turned out to be a bug in the FusionRenderer.
I opened a ticket and pushed a fix for that: https://github.com/neos/form-fusionrenderer/pull/11

**Comment by Sebobo** [comment]
Then I would assume the rest is fine. I couldn't find any other problem yet.

**Pull request merged**